GameOfRunes/GameOfRunes/Levels and Waves/EnemySpawnUnit.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  EnemySpawnUnit.swift
+//  GameOfRunes
+//
+//  Created by Brian Yen on 28/3/20.
+//  Copyright © 2020 TeamHoWan. All rights reserved.
+//
+
+/**
+ This struct is used as the basic unit for creating a level. Different instances
+ of it can be concatenated together to create different types of levels. See
+ `LevelCreator` for more examples on how `EnemySpawnUnit` can
+  be used to create interesting levels.
+
+ - Note: The position of the `EnemyType` within the spawn wave (i.e. the [EnemyType]
+ in `unit` will determine the spawn position of the enemy (which lane the
+ enemy will be spawned at).
+ */
+struct EnemySpawnUnit {
```

Comment:
The implementation of `EnemySpawnUnit` doesn't seem like it is a small unit - it pretty much represents the entire level. I'm not sure if that's what you're going for?